src/index.js

```diff
@@ -19,60 +19,112 @@ const commands = Object.keys(queries).map(queryName => {
-            doc.querySelector('.___cypressNotExistingSelector'),
-          )
-      } else {
-        commandImpl = doc => baseCommandImpl(doc)
+      let commandImpl = doc => baseCommandImpl(doc)
+
+      var inputArr = args.filter(filterInputs);
+
+      var consoleProps = {
+        // TODO: Would be good to completely separate out the types of input into their own properties
+        input: inputArr
       }
-      const thenHandler = new Function(
-        'commandImpl',
-        `
-            return function Command__${queryName}(thenArgs) {
-              return commandImpl(thenArgs.document)
-            }
-          `,
-      )(commandImpl)
+
+      Cypress.log({
+        $el: inputArr,
+        name: queryName,
+        message: inputArr,
+        consoleProps: () => consoleProps
+      });
+
       return cy
         .window({log: false})
-        .then({timeout: waitOptions.timeout + 100}, thenHandler)
-        .then(subject => {
-          Cypress.log({
-            $el: subject,
-            name: queryName,
-            message: args.filter(value => {
-              if (Array.isArray(value) && value.length === 0) {
-                return false
-              }
-              if (value instanceof RegExp) {
-                return value.toString()
+        .then((thenArgs) => {
+          const getValue = () => {
+            var result;
+            try {
+              var value = commandImpl(thenArgs.document);
+              result = Cypress.$(value);
+            }
+            catch (err) {
+              // Catch exceptions where it can't find the element, so we can retry
+              if (/Unable to find an element with the text/.test(err.message))
+              {
+                consoleProps.error = err;
+                result = Cypress.$();
+              } else {
+                throw err;
               }
-              if (
-                typeof value === 'object' &&
-                Object.keys(value).length === 0
-              ) {
-                return false
+            }
+          
+            // Overriding the selector of the jquery object because it's displayed in the long message of .should('exist') failure message
+            // Hopefully it makes it clearer, because I find the normal response of "Expected to find element '', but never found it" confusing
+            result.selector = queryName + '(' + findTextOrRegex(args) + ')';
+
+            if (result.length > 0) {
+              consoleProps.yielded = result.toArray()
+            }
+
+            return result;
+          }
+
+          const resolveValue = () => {
+            // retry calling "getValue" until following assertions pass or this command times out
+            return Cypress.Promise.try(getValue).then(value => {
+              return cy.verifyUpcomingAssertions(value, waitOptions, {
+                onRetry: resolveValue,
+              })
+            })
+          }
+
+          if (/queryBy|queryAllBy/.test(queryName)) {
```

Comment:
Yes that's the only difference at the moment. We can't retry the something where it's valid to return a null or empty array, simply because the retry-ability will keep going until it finds something.

I simply run the function to get the dom element and return early, no retryability for these. The only other option would be to keep retrying until **just** before the original timeout, but to me that would slow down tests unnecessarily, the developer should check the page is ready through some other query before using query*

Reply:
Sounds like a reasonable change. I don't think we need to support all the same queries for Cypress if they don't make sense with the Cypress ecosystem

Reply:
There were no tests for find* so out of curiosity I've converted all of the get* tests to find* locally to see what happens, of which they all fail.

I've only ever used the get* and query* queries, and hadn't realised the implications of find*.

I've quickly tried to fix the find* queries, however I've found that when it can't find an element it doesn't show the spinner because it's still waiting for a response from the find* queries.

The try-ability code works on the premise that it will find the value, check to see if the next assertion passes and if not try again, until the timeout expires. Due to the fact it only tries once, we don't see the spinner.

I don't mind get* or find* are removed, but either way they both need to be implemented using the get* underlying functions.

Reply:
I still think the query* is a valid use case, where as get and find are interchangeable in the case of Cypress.

Reply:
Using the `query*` implementations for both `find*` and `query*` would mean we no longer need changes to DTL.

`find*` can use the `query*` implementations so it can keep retrying until results are found. `query*` won't throw errors in the scenario of nothing found, which suits us perfect.

`query*` can return found item(s) immediately, which is the trade off for warranting the ability to have no results found.

`get*` feels unnecessary now because even if we return nothing immediately, it still takes the timeout duration before it fails the assertion. Additionally the spinner will no longer appear due to the command not being re-run, which I feel adds to confusion because the test runs for a little longer with no visual feedback as to why.

It's likely these changes will need a cheatsheet created to help explain the query differences between DTL and CTL.

Reply:
You could use either really, although findByText would be the more robust option.

For example if you were deleting an item from a list and it disappeared after a fade out of 0.5 seconds, queryByText would fail because it would find the dom item immediately. Where as findByText would keep trying until the assertion passed or it timed out.

Reply:
I get what you mean, it would be more inline with Cypress' methodology. You can use `.should('have.length', 1)` to assert you only have one result after all. On top of that I believe if you return multiple results it would fail on commands like `click()` anyway.

If we did create only one variant of `query*` and `find*` would it be named with or without the *All*? If it's without the *All* then the functionality diverges away from the DTL query conventions drastically. If you were going to break from the DTL conventions I'd probably recommend going with the `get*` names, as it lines up nicely in Cypress.

That being said I worry about moving too far away from DTL conventions, and what has been this library's API up until now. I feel like implementing `query*`, `queryAll*`, `find*` and `findAll*` stays true to DTL while keeping all the functionality usable within the shell of Cypress, with all the benefits that brings.

Happy to amend as necessary once you have decided what you prefer.

Reply:
To confirm, are you both saying you are happy for CTL to have the 4 queries?

- queryBy*
- queryAllBy*
- findBy*
- findAllBy*

Reply:
Not at all, I'm doing very little in the grand scheme of things. We started using this library at work, and I saw the issue and realised I could do something about it.

I need to get something else done, but should be able to do the work over the weekend / early next week.

Reply:
Super excited to get this merged! I will see if I can use your repo at my work's project to improve the tests and let you know how it goes. Currently, I have a bunch of wait() to give the page time to render some elements. 

My understanding is that with this PR I could avoid these changes. I also it sounds like it should use findByText etc instead of getByText. 

How can I help getting this merged?

Reply:
Sorry for the delay guys, things have been hectic at work and at home. I will get onto it soon.